crates/core/src/node/network_bridge/p2p_protoc.rs

```diff
@@ -349,13 +349,60 @@ impl P2pConnManager {
                                 })??;
                             }
                             NodeEvent::QuerySubscriptions { callback } => {
-                                // For now, return connection info until we integrate with contract handler
-                                // TODO: Query contract handler for actual subscription info
+                                // Get network subscriptions from OpManager
+                                let network_subs = op_manager.get_network_subscriptions();
+
+                                // Get application subscriptions from contract executor
+                                // For now, we'll send a query to the contract handler
+                                let (tx, mut rx) = tokio::sync::mpsc::channel(1);
+
+                                op_manager
+                                    .notify_contract_handler(
+                                        ContractHandlerEvent::QuerySubscriptions { callback: tx },
+                                    )
+                                    .await?;
+
+                                let app_subscriptions =
+                                    match timeout(Duration::from_secs(1), rx.recv()).await {
+                                        Ok(Some(QueryResult::NetworkDebug(info))) => {
+                                            info.subscriptions
+                                        }
+                                        _ => Vec::new(),
+                                    };
+
+                                // Combine network and application subscriptions
+                                let mut all_subscriptions = app_subscriptions;
```

Comment:
for debugging pourpouses I think it would make sense to keep them clearly separated

Reply:
You're right about keeping them separated for debugging. I'm updating the `NetworkDebugInfo` structure to have separate fields:
- `application_subscriptions`: WebSocket clients subscribed to contracts  
- `network_subscriptions`: Nodes subscribing for routing purposes

This will make it much clearer when debugging which type of subscription we're looking at.